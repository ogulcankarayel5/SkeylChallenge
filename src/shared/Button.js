import styled from 'styled-components'


export const Button = styled.input.attrs((props)=>({
    type:props.type

  
}))`
grid-area:"button";
letter-spacing: 0.1rem;
  font-size: ${(props) => props.fontSize ?? "2rem"};
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.background};
  background-color: ${(props) =>
    props.backgroundColor
      ? props.backgroundColor
      : props.theme.colors.formButtonColor};
  padding: ${(props) => props.padding ?? "2rem"};
  border: 2px solid ${(props) => props.borderColor};
  font-weight: ${(props) => props.fontWeight ?? 500};
  
  cursor: pointer;
 

`