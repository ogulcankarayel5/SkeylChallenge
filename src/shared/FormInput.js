import React from 'react'
import {Input} from '.'


export const FormInput = React.memo(({name,type,placeholder,value,onChange,className,ariaLabel}) => {
 
  return (
      <>
      <Input data-test="Input" className={className} name={name} type={type} placeholder={placeholder} value={value} onChange={onChange} aria-label={ariaLabel}/>
      
      
      </>
  )
})