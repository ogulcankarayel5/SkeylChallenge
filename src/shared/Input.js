import styled from "styled-components";

export const Input = styled.input.attrs((props) => ({
  type: props.type,
}))`
grid-area:"input";
  border-radius: 4px;
  font-weight: 600;
  letter-spacing: 0.1rem;
  font-family: "Open Sans";
  font-size: 1.5rem;
  

  border: 2px solid #aaa;
  padding: 2rem;

  ::placeholder {
    font-weight: 400;
    opacity: 0.9;
  }
  &:focus {
    border-color: ${(props) => props.theme.colors.formInputBorderColor};
    box-shadow: 0 0 8px 0 ${(props) => props.theme.colors.formInputBorderColor};
  }
`;
