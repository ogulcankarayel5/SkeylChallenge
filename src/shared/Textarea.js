import styled from 'styled-components'

export const Textarea=styled.textarea`
    grid-area:"textarea";
    margin-top:${props=>props.marginTop ?? props.marginTop}

`