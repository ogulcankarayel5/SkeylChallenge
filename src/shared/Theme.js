
import React from "react";
import { ThemeProvider } from "styled-components";

 export const theme = {
  colors: {
    primary:"#00C4EA",
    border: "blue",
    background: "#FFFFFF",
    formButtonColor:"#7041EE",
    
    
    
  }
 
 
};

export const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);