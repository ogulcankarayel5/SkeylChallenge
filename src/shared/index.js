export * from './Button'
export * from './Input'
export * from './Textarea'
export * from './Theme'
export * from './FormInput'
