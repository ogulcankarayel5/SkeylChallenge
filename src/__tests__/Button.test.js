import React from 'react'

import {Button} from '../shared'
import {theme} from '../shared'
import renderer from 'react-test-renderer'
import 'jest-styled-components'



describe("Button snapshot",() => {
   
    it("should match snapshot",() => {
        const tree = renderer.create(<Button theme={theme}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
}) //snapshot s覺k覺nt覺l覺