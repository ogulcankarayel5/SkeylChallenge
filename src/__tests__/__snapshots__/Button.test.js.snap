// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot should match snapshot 1`] = `
.c0 {
  grid-area: "button";
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  font-size: 2rem;
  color: #FFFFFF;
  background-color: #7041EE;
  padding: 2rem;
  border: 2px solid;
  font-weight: 500;
  cursor: pointer;
}

<input
  className="c0"
/>
`;
